### ISSUE TITLE
`DeprecationWarning` appearing in logs after running action

### ISSUE THREAD

**Issue by tmillr** [peter-evans/create-pull-request#1336]
```
(node:1790) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[17](https://github.com/tmillr/breaking-change/actions/runs/3607545212/jobs/6079626552#step:7:18)
(Use `node --trace-deprecation ...` to show where the warning was created)
```

https://github.com/tmillr/breaking-change/actions/runs/3607545212/jobs/6079626552#step:7:17

Perhaps a dependency needs updating?

**Comment by matanshk** [comment]
same here.
any updates?

**Comment by tmillr** [comment]
@peter-evans

**Comment by peter-evans** [comment]
It seems that this warning started when I introduced the `proxy-agent` dependency in https://github.com/peter-evans/create-pull-request/commit/b4d51739f96fca8047ad065eccef63442d8e99f7

More investigation is needed to figure out where in the dependency chain the deprecated methods are being used. I'll look into this when I have more time. Or if anyone would like to investigate that would be very helpful.

**Comment by peter-evans** [comment]
I'm planning to remove the `proxy-agent` dependency with a new proxy implementation. This warning will go away then. The new implementation is currently sitting in the `v5-beta` branch, which I'm planning to release as a new major version at some point in the near future.

**Comment by khajavi** [comment]
I have the same problem

**Comment by IvanPizhenko** [comment]
Facing the same issue with latest v4.

**Comment by peter-evans** [comment]
If the warning is annoying feel free to use the `v5-beta` version of the action. If you use it, keep subscribed to this issue and I'll make sure there is plenty of warning before that version is deleted after the `v5` release.

```yml
      - uses: peter-evans/create-pull-request@v5-beta
```

**Comment by IvanPizhenko** [comment]
Thank you, Peter, actually it is not so annoying, but just I faced it. I think I can wait for the final v5. Please post here a message when it is out.

**Comment by khajavi** [comment]
@peter-evans Thank you for the update!
So, I'm looking forward the next release.

**Comment by peter-evans** [comment]
This is now fixed in `v5.0.0` / `v5`.

If you are using `v5-beta` or `v5-rc`, please update to `v5`. I will remove pre-release versions after 4 weeks. Thank you for testing the pre-release version!
